diff -r d241b80712dc unsupported/Eigen/CXX11/src/Tensor/TensorDeviceSycl.h
--- a/unsupported/Eigen/CXX11/src/Tensor/TensorDeviceSycl.h	Fri Mar 09 11:31:25 2018 +0000
+++ b/unsupported/Eigen/CXX11/src/Tensor/TensorDeviceSycl.h	Fri Mar 16 13:36:09 2018 +0000
@@ -26,7 +26,7 @@
   #define ConvertToActualSyclOffset(Scalar, offset) (((offset) * sizeof(buffer_scalar_t)) / sizeof(Scalar))
 
 typedef cl::sycl::codeplay::buffer_data_type_t buffer_scalar_t;
-typedef std::allocator<buffer_scalar_t> buffer_alloc_t;
+typedef cl::sycl::default_allocator<buffer_scalar_t> buffer_alloc_t;
 
 template <cl::sycl::access::mode AcMd>
 struct RangeAccess {
diff -r d241b80712dc unsupported/Eigen/CXX11/src/Tensor/TensorSyclVirtualPointer.h
--- a/unsupported/Eigen/CXX11/src/Tensor/TensorSyclVirtualPointer.h	Fri Mar 09 11:31:25 2018 +0000
+++ b/unsupported/Eigen/CXX11/src/Tensor/TensorSyclVirtualPointer.h	Fri Mar 16 13:36:09 2018 +0000
@@ -54,7 +54,7 @@
 /**
  * Default values for template arguments
  */
-using buffer_data_type_t = uint8_t;
+using buffer_data_type_t = float;
 using buffer_allocator_base_t = cl::sycl::buffer_allocator<buffer_data_type_t>;
 using buffer_allocator_default_t =
     cl::sycl::default_allocator<buffer_data_type_t>;
